### PR TITLE
Multi project

### PR DESCRIPTION
**Pull request by kensipe**
large change to build as a multi-project build.  It was too much work to rebase the previous [PR-77](https://github.com/mesos/myriad/pull/77)

This is basically the same work however I've updated all the docs now and have tested the vagrant configuration.  

Lets get this checked and merged!

### CODE REVIEW

**Comment by kensipe** [comment]
@DarinJ @mohitsoni @adam-mesos @smarella  any review here.   I would like to merge this, this week.  Tomorrow if possible.

**Comment by DarinJ** [comment]
Should we distribute:  gradle/wrapper/gradle-wrapper.jar?

**Comment by kensipe** [comment]
no... that is not part of the distro... only used for the build process

**Comment by kensipe** [comment]
several people seem confused by the gradle-wrapper.jar and the fact that we have a binary checked into the repo.  It is idiomatic gradle code to include it.  it is a 25k jar that is a bootstrap jar for pulling the correct version of gradle for the build.  It ensures that builds **always** work because the meta-data for what the correct tool and version for building is included with the project.

File: docs/myriad-remote-distribution-configuration.md [hunk 1]
```diff
@@ -12,29 +12,35 @@ your installation.
...
-```Shell
-hadoop fs -put build/libs/myriad-executor-0.0.1.jar /dist
+cp myriad-scheduler/build/libs/*.jar /opt/hadoop-2.5.0/share/hadoop/yarn/lib/
+cp myriad-scheduler/src/main/resources/myriad-config-default.yml /opt/hadoop-2.5.0/share/hadoop/yarn/lib/
+```
+
+You will also need to place `myriad-executor-runnable-0.0.1.jar` in hdfs
+
 ```
+hadoop fs -put myriad-executor/build/libs/myriad-executor-runnable-0.0.1.jar /dist
```
Comment (DarinJ) [inline docs/myriad-remote-distribution-configuration.md:31]: probably should be consistant with myriad-executor-runnable-0.0.1.jar and myriad-executor-runnable-xxx.jar throughout documentation.  That's not on you though.

File: myriad-commons/src/main/java/com/ebay/myriad/executor/MyriadExecutorDefaults.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
+package com.ebay.myriad.executor;
+
+/**
+ * Myriad's Executor Defaults
+ */
+public interface MyriadExecutorDefaults {
```
Comment (DarinJ) [inline myriad-commons/src/main/java/com/ebay/myriad/executor/MyriadExecutorDefaults.java]: Interface vs Static Class?

**Comment by DarinJ** [comment]
During build two warnings about style.  +1 for warnings instead of errors, but would be nice if Travis would still complain.
First look no deal breakers, one comment on style.  Will look once more tomorrow morning.

File: myriad-commons/src/main/java/com/ebay/myriad/executor/MyriadExecutorDefaults.java (hunk 2 above)
Comment (kensipe) [inline myriad-commons/src/main/java/com/ebay/myriad/executor/MyriadExecutorDefaults.java]: roger... switching to class.

File: docs/myriad-remote-distribution-configuration.md (hunk 1 above)
Comment (kensipe) [inline docs/myriad-remote-distribution-configuration.md:31]: ok... I've updated several.    here is the rule I followed:  1) if it was documentation / mentioned in docs,  it has myriad-executor-runnable-x.x.x.jar, however if 2) it was used in a config file example like yml or part of a shell example I left the myriad-executor-runnable-0.0.1.jar version.

**Comment by kensipe** [comment]
regarding warnings.  I was trying to get this PR in with min changes to the project... just the structure.  however I will take a look and fix these.

**Comment by kensipe** [comment]
done!    @DarinJ what do you think... are we good to merge ?

**Comment by pdread100** [comment]
+1

**Comment by DarinJ** [comment]
@kensipe if you've tested it, I think we're good.  I got through the build on a cluster last night, but haven't had a chance to install.  I saw no other issues today.

**Comment by kensipe** [comment]
I did some vagrant testing... I think we are good to go!

**Pull request merged**